src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -6666,7 +6665,14 @@ public IContentReader GetContentReader(string path)
 
             try
             {
-                if (Directory.Exists(path))
+                // Get-Content will write a non-terminating error if the target is a directory.
+                // On Windows, the streamName must be null or empty for it to write the error. Otherwise, the
+                // alternate data stream is not a directory, even if it's set on a directory.
+                if (Directory.Exists(path)
+#if !UNIX
+                    && string.IsNullOrEmpty(streamName)
+#endif
```

Comment:
The duplication of both the code and the comment implies that a boolean method might be a good way to go here

Reply:
The variable `streamname` doesn't exist on UNIX platforms.  As such, it would require an `#if/#endif` block in both places anyway, and require processing a semantic streamname variable on UNIX where care has been taken to avoid processing such a semantic.  I agree with you in principle, but in practice it would be even messier.  (The signature would be `bool isActualContainer(string path, string streamName=null)`, but the call would need to be
```cs
                if (isActualContainer(path
#if !UNIX
                                                         ,streamName
#endif
                   )
```
and the method would be
```cs
bool isActualContainer(string path, string streamName=null)
{
    return (Directory.Exists(path) && string.IsNullOrEmpty(streamName);
}
```
I was unwilling to introduce such semantic processing on non-Windows platforms, because care has already been taken to prevent such.

I mean, I'll do it if you still want me to, but it feels like an antipattern here.

Reply:
Agreed that the ifdef complicates it more than it's worth